In [1]:
import pandas as pd

In [35]:
#loading the dataset
df=pd.read_csv(r"C:\Users\rajan\Downloads\archive (2)\Housing.csv")

In [36]:
#Display the first few rows of the DataFrame to understand its structure.
df.head()

price  area  bedrooms  bathrooms  stories mainroad guestroom basement  \
0  13300000  7420         4          2        3      yes        no       no   
1  12250000  8960         4          4        4      yes        no       no   
2  12250000  9960         3          2        2      yes        no      yes   
3  12215000  7500         4          2        2      yes        no      yes   
4  11410000  7420         4          1        2      yes       yes      yes   

  hotwaterheating airconditioning  parking prefarea furnishingstatus  
0              no             yes        2      yes        furnished  
1              no             yes        3       no        furnished  
2              no              no        2      yes   semi-furnished  
3              no             yes        3      yes        furnished  
4              no             yes        2       no        furnished

In [37]:
#summary statistics
df.describe()

price          area    bedrooms   bathrooms     stories  \
count  5.450000e+02    545.000000  545.000000  545.000000  545.000000   
mean   4.766729e+06   5150.541284    2.965138    1.286239    1.805505   
std    1.870440e+06   2170.141023    0.738064    0.502470    0.867492   
min    1.750000e+06   1650.000000    1.000000    1.000000    1.000000   
25%    3.430000e+06   3600.000000    2.000000    1.000000    1.000000   
50%    4.340000e+06   4600.000000    3.000000    1.000000    2.000000   
75%    5.740000e+06   6360.000000    3.000000    2.000000    2.000000   
max    1.330000e+07  16200.000000    6.000000    4.000000    4.000000   

          parking  
count  545.000000  
mean     0.693578  
std      0.861586  
min      0.000000  
25%      0.000000  
50%      0.000000  
75%      1.000000  
max      3.000000

In [49]:
#Q1

df.shape

(545, 13)

In [39]:
#Q2

df.dtypes

price                int64
area                 int64
bedrooms             int64
bathrooms            int64
stories              int64
mainroad            object
guestroom           object
basement            object
hotwaterheating     object
airconditioning     object
parking              int64
prefarea            object
furnishingstatus    object
dtype: object

In [40]:
#Q3

df.isna().sum()
#there is no missing values in the dataset

price               0
area                0
bedrooms            0
bathrooms           0
stories             0
mainroad            0
guestroom           0
basement            0
hotwaterheating     0
airconditioning     0
parking             0
prefarea            0
furnishingstatus    0
dtype: int64

In [42]:
#Q4

df['furnishingstatus'].unique()

array(['furnished', 'semi-furnished', 'unfurnished'], dtype=object)

In [43]:
df['furnishingstatus'].nunique()

3

In [44]:
#Q5

mean_price=df['price'].mean()
median_price=df['price'].median()
std_price=df['price'].std()
print(f"mean : {mean_price}\nmedian : {median_price}\nstd : {std_price}")

mean : 4766729.247706422
median : 4340000.0
std : 1870439.615657394


In [52]:
#Q6

Q1 = df['area'].quantile(0.25)
Q3 = df['area'].quantile(0.75)
IQR = Q3 - Q1
min_value = Q1 - 1.5 * IQR
max_value = Q3 + 1.5 * IQR
outliers = df[(df['area'] < lower) | (df['area'] > upper)]
print(f"Number of outliers: {outliers.shape[0]}")
print(f"Outliers: \n{outliers['area']}")

Number of outliers: 12
Outliers: 
7      16200
10     13200
56     11440
64     11175
66     13200
69     12090
125    15600
129    11460
186    11410
191    10700
211    12900
403    12944
Name: area, dtype: int64


Calculated max and min value using IQR,Q1 and Q2.
If the value is greater than max value or smaller than min value then there is outlier.
Here 12 outliers are there.

In [47]:
#Q7

df.corr()

price      area  bedrooms  bathrooms   stories   parking
price      1.000000  0.535997  0.366494   0.517545  0.420712  0.384394
area       0.535997  1.000000  0.151858   0.193820  0.083996  0.352980
bedrooms   0.366494  0.151858  1.000000   0.373930  0.408564  0.139270
bathrooms  0.517545  0.193820  0.373930   1.000000  0.326165  0.177496
stories    0.420712  0.083996  0.408564   0.326165  1.000000  0.045547
parking    0.384394  0.352980  0.139270   0.177496  0.045547  1.000000

Here price is moderately correlated with area and bathromms.Therefore houses that have larger areas and more bathrooms have highest prices. 
price is less correlated to bedrooms and parking

In [48]:
#Q8

df.groupby('furnishingstatus').agg({'price':['mean','sum']})

price            
                          mean         sum
furnishingstatus                          
furnished         5.495696e+06   769397440
semi-furnished    4.907524e+06  1114008000
unfurnished       4.013831e+06   714462000

#Q9

From the housing price dataset analysis, dataset consists of 545 rows and 13 columns. 
Average price of house is approximately 4.8 million.
Price of house is moderately correlated to area and bathrooms which indicate that houses that have larger area and more bathrooms have highest prices. 
Average price is high for furnished houses compared to semi-furnished and unfurnished houses